scripts/publish_release_to_aws.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+declare -A moduleToStorageIdMap
+moduleToStorageIdMap["libnavigation-base"]="mobile-navigation-base"
+moduleToStorageIdMap["libnavigation-core"]="mobile-navigation-core"
+moduleToStorageIdMap["libnavigation-metrics"]="mobile-navigation-metrics"
+moduleToStorageIdMap["libnavigator"]="mobile-navigation-navigator"
+moduleToStorageIdMap["libtrip-notification"]="mobile-navigation-notification"
+moduleToStorageIdMap["libdirections-hybrid"]="mobile-navigation-router"
+moduleToStorageIdMap["libdirections-offboard"]="mobile-navigation-router-offboard"
+moduleToStorageIdMap["libdirections-onboard"]="mobile-navigation-router-onboard"
+moduleToStorageIdMap["libnavigation-util"]="mobile-navigation-utils"
+moduleToStorageIdMap["libnavigation-ui"]="mobile-navigation-ui-v1"
+
+function check_file_exists() {
+  if [ ! -f $1 ]; then
+    echo "$1 does not exist."
+    exit 1
+  fi
+}
+
+function get_s3_storage_path() {
+  repo_type="releases"
+  if [ $1 == true ]; then
+    repo_type="snapshots"
+  fi
+  echo "s3://mapbox-api-downloads-production/v2/${moduleToStorageIdMap[$2]}/$repo_type/android/$3/maven/$4"
+}
+
+module="$1"
+artifact_id="$2"
+version="$3"
+variant="$4"
+if [ -z "$module" ]; then
+  echo >&2 "First argument (module name) is missing."
+  exit 1
+fi
+if [ -z "$artifact_id" ]; then
+  echo >&2 "Second argument (Maven artifact ID) is missing."
+  exit 1
+fi
+if [ -z "$version" ]; then
```

Comment:
should we have a regex to check version?

Reply:
We shouldn't enforce it on this step. There's the telem plugin mentioned above that does check it already. Here, we could hardcode a snapshot version that doesn't follow semver.